lib/gc.cpp

```diff
@@ -33,26 +34,57 @@ limitations under the License.
 
 // One can disable the GC, e.g., to run under Valgrind, by editing config.h
 #if HAVE_LIBGC
-static bool done_init;
+static bool done_init, started_init;
 void *operator new(std::size_t size) {
     /* DANGER -- on OSX, can't safely call the garbage collector allocation
      * routines from a static global constructor without manually initializing
      * it first.  Since we have global constructors that want to allocate
      * memory, we need to force initialization */
     if (!done_init) {
+        started_init = true;
         GC_INIT();
         done_init = true; }
     return ::operator new(size, UseGC, 0, 0);
 }
 void *operator new[](std::size_t size) {
     if (!done_init) {
+        started_init = true;
```

Comment:
why not use `std::call_once`?

Reply:
`call_once` doesn't work in the recursive call case -- we need to detect if we've started to do GC_INIT but it has not yet finished.